src/Kovan-DssSpell.t.sol

```diff
@@ -1278,71 +1698,390 @@ contract DssSpellTest is DSTest, DSMath {
+        assertEq(vow.wards(address(end_old)), 0);
+        assertEq(pot.wards(address(end_old)), 0);
+        assertEq(spotter.wards(address(end_old)), 0);
+
+        assertEq(vat.wards(address(end)), 1);
+        assertEq(cat.wards(address(end)), 1);
+        assertEq(dog.wards(address(end)), 1);
+        assertEq(vow.wards(address(end)), 1);
+        assertEq(pot.wards(address(end)), 1);
+        assertEq(spotter.wards(address(end)), 1);
+
+        assertEq(end.wards(address(esmBug)), 1);
+        assertEq(end.wards(address(esmAttack)), 1);
+        assertEq(vat.wards(address(esmAttack)), 1);
+        assertEq(end_old.wards(address(esm_old)), 1);
+    }
 
-        if (spell.officeHours()) {
-            hevm.warp(monday_1400_UTC - 1 days);                       // Sunday,   14:00 UTC
-            assertEq(spell.nextCastTime(), monday_1400_UTC);           // Monday,   14:00 UTC
+    function testSpellIsCast_new_End_functionality() public {
+        vote(address(spell));
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
 
-            hevm.warp(monday_1400_UTC - 2 days);                       // Saturday, 14:00 UTC
-            assertEq(spell.nextCastTime(), monday_1400_UTC);           // Monday,   14:00 UTC
+        DSTokenAbstract ETH = DSTokenAbstract(addr.addr("ETH"));
+        GemJoinAbstract joinETHA = GemJoinAbstract(addr.addr("MCD_JOIN_ETH_A"));
+        FlipAbstract flipETHA = FlipAbstract(addr.addr("MCD_FLIP_ETH_A"));
+
+        DSTokenAbstract BAT = DSTokenAbstract(addr.addr("BAT"));
+        GemJoinAbstract joinBATA = GemJoinAbstract(addr.addr("MCD_JOIN_BAT_A"));
+        FlipAbstract flipBATA = FlipAbstract(addr.addr("MCD_FLIP_BAT_A"));
+
+        DSTokenAbstract LINK = DSTokenAbstract(addr.addr("LINK"));
+        GemJoinAbstract joinLINKA = GemJoinAbstract(addr.addr("MCD_JOIN_LINK_A"));
+        ClipAbstract clipLINKA = ClipAbstract(addr.addr("MCD_CLIP_LINK_A"));
+
+        uint256 ilkAmt = 1 * THOUSAND * WAD;
+
+        giveTokens(ETH, ilkAmt);
+        giveTokens(BAT, ilkAmt);
+        giveTokens(LINK, ilkAmt);
+
+        ETH.approve(address(joinETHA), ilkAmt);
+        joinETHA.join(address(this), ilkAmt);
+        BAT.approve(address(joinBATA), ilkAmt);
+        joinBATA.join(address(this), ilkAmt);
+        LINK.approve(address(joinLINKA), ilkAmt);
+        joinLINKA.join(address(this), ilkAmt);
+
+        (,uint256 rate, uint256 spot,,) = vat.ilks("ETH-A");
+        vat.frob("ETH-A", address(this), address(this), address(this), int256(ilkAmt), int256(mul(ilkAmt, spot) / rate));
+        (, rate, spot,,) = vat.ilks("BAT-A");
+        vat.frob("BAT-A", address(this), address(this), address(this), int256(ilkAmt), int256(mul(ilkAmt, spot) / rate));
+        (, rate, spot,,) = vat.ilks("LINK-A");
+        vat.frob("LINK-A", address(this), address(this), address(this), int256(ilkAmt), int256(mul(ilkAmt, spot) / rate));
+
+        hevm.warp(block.timestamp + 1);
```

Comment:
It's theoretically possible that the interest accrued in 1 block is less than the rounding error...might bump this to `+ 100` or something just to minimize any chance of flakiness.

Reply:
Yeah I was wondering if some day that would happen. We have been using this + 1 for a long time and so far it has been working without any issue. I like the +1 because is very explicit that we put the vault right to the limit, but will consider changing it to something larger to avoid a possible rounding error.